### DISCUSSION TITLE
Lock Supabase Studio Table Editor

### DISCUSSION THREAD

**Discussion by mclean25** [supabase/supabase#12432] [Feature Requests]
# Feature request
Add a setting to prevent any edits to the schema via the table editor in Supabase Studio

## Is your feature request related to a problem? Please describe.
If you're developing Supabase locally and are creating migrations to change the "production" environment (via `supabase db push`), things can get really messed up if someone happens to edit the schema via the table editor. This is exacerbated by the fact that Supabase Studio looks pretty similar between `app.supabase.io` and `localhost`, so one could accidentally modify things thinking it's the local editor. 

## Describe the solution you'd like
I would think a boolean setting preventing changes to the Table Editor would suffice. v2 could be "admins only" or something.

**Comment by saltcod** [comment]
Great suggestion and one we think could be possible to implement with user roles down the road. I'm going to note it in our internal backlog as something to consider down the road. 

Appreciate the feedback!

**Comment by MildTomato** [comment]
This is a part duplicate of https://github.com/orgs/supabase/discussions/16759, but we'll keep both open since they both have 1 upvote and cover different use cases.

**Comment by pythonicode** [comment]
Any updates on this? Would love to be able to lock my tables in prod from being edited in the dashboard

**Comment by kiptoomm** [comment]
Any update on this @saltcod? I have experienced major pain resolving divergence between our db and migrations files because a collaborator on the team inadvertently made several changes on the dashboard

**Comment by kbrgl** [comment]
+1 on this issue, this is a major security hole in production!

**Comment by moe12572** [comment]
+1 on this, how does this feature not exist esp when you are dealing with prod data?

**Comment by akashagarwal7** [comment]
+1

**Comment by jonathanlab** [comment]
+1

**Comment by mikkelsvartveit** [comment]
+1, this is badly needed for the local development setup you recommend in the docs

**Comment by Jonny1987** [comment]
+1 for this issue. There needs to be a safeguard in place to stop accidental changes to production in the dashboard
